### PR TITLE
#160452497 Role based access control functionality

### PR DESCRIPTION
**Pull request by Alameen688**
#### What does this PR do?
Implements role-based access control by creating a middleware that checks if the logged in user has the specified role.
#### Description of Task to be completed?
- Add test suites for access control
- Add a class that guards route by receiving the role that is allowed
- Add an endpoint to disable article, that can be accessed only by an admin
- Add Sequelize seeder for initializing the database with dummy data.
- Add pull request template
- Update package.json file
#### How should this be manually tested?
- Run `git fetch` to fetch the new branch(es)
- Checkout to this branch `git checkout ft-160452497-access-control`
- Run `npm install` to install dependencies
- Run migrations `npm run migrations`
- Run seeders `npm run seed`
- Run `npm start` to start the application
- Run `npm test` to run tests
- [see docs](http://localhost:3000/api-docs/)
- [PUT](http://localhost:3000/api-docs/#/Admin/put_articles__slug__disable) `/api/v1/articles/{slug}/disable`
Log in with an admin account
```
{
  email: 'admin@local.host',
  password: 'administrator'
}
```
and use the token to disable any article of your choice
Try using any user who is not the owner of an article you would like to disable. This should result in a 401 response
- [PUT](http://localhost:3000/api-docs/#/Super_Admin/put_users__id___role_) `/api/v1/users/{userId}/{role}`
As a superadmin, the above endpoint can be used to change the role of a user.
#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
[#160452497](https://www.pivotaltracker.com/story/show/160452497)
#### Questions:

### CODE REVIEW

File: helpers/ArticleHelper.js [hunk 1]
```diff
@@ -131,13 +131,12 @@ class ArticleHelper {
...
    */
   static getArticlesResponseData(allArticle) {
-    const articles = [];
-    allArticle.forEach((article) => {
-      const { user, tags } = article;
-      articles.push(ArticleHelper.getArticleResponseData(user,
-        article, tags));
-    });
-    return articles;
+    return allArticle.filter(article => !article.disabled)
```
Comment (omedale) [inline helpers/ArticleHelper.js]: Adjust this please

File: tests/middlewares/Guard.test.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import chai from 'chai';
```
Comment (omedale) [inline tests/middlewares/Guard.test.js]: Remove this file please

File: helpers/ArticleHelper.js (hunk 1 above)
Comment (Alameen688) [reply helpers/ArticleHelper.js]: Alright, done.

File: tests/middlewares/Guard.test.js (hunk 2 above)
Comment (Alameen688) [reply tests/middlewares/Guard.test.js]: I removed the file at some point. However, my test coverage dropped. The file contains unit tests for the Guard middleware. Thank you.

**Comment by olusoladavid** [comment]
Great job on this feature @Alameen688 . I especially liked how you used OOP patterns in the implementation. I have reviewed this pull request to ensure it complies with the team's coding standards and the acceptance criteria of the associated story card. I approve it for merging.

**Comment by omedale** [comment]
Cool

**Pull request merged**